### ISSUE TITLE
Updating ViewModel cache.

### ISSUE THREAD

**Issue by dangreen** [ractivejs/ractive#2316]
In some situation i get this:
```js
temp1.get().data.items[4].name.value
>>> "Member Name!!!"
temp1.get("data.items[4].name.value")
>>> "Member Name"
```

How i can manually update cache?

**Comment by Madgvox** [comment]
Ractive has two update methods, [`ractive.update()`](http://docs.ractivejs.org/latest/ractive-update) and [`ractive.updateModel()`](http://docs.ractivejs.org/latest/ractive-updatemodel). `update` is useful if you update data without using the built in setter methods, while `updateModel` is useful if you programmatically manipulate inputs and other elements that have two‐way binding set up.

**Comment by dangreen** [comment]
@Madgvox This methods doesn't updates `data.items[4].name.value` value.
```js
temp1.get("data.items[4].name.value")
"Member Name"
temp1.get().data.items[4].name.value
"Member Name!"
temp1.update()
temp1.get("data.items[4].name.value")
"Member Name"
temp1.updateModel()
temp1.get("data.items[4].name.value")
"Member Name"
```

**Comment by Madgvox** [comment]
What version of ractive are you on & can you post a jsfiddle repro? That could be a bug.

**Comment by dangreen** [comment]
@Madgvox 0.7.3 http://jsfiddle.net/ddgamw8L/4/ . I know, code is very unusual :-)

**Comment by martypdx** [comment]
The bugs in the `ArrayAdaptor`. [It works](http://jsfiddle.net/6xhoh2vn/) if you use the array api method instead:

```
this.push("context.data.value.items", {});
```

**Comment by dangreen** [comment]
@martypdx https://github.com/ractivejs/ractive/issues/2275 "Use the array methods instead of api method" :-|

**Comment by dangreen** [comment]
@martypdx but thanks, anyway...

**Comment by martypdx** [comment]
> @martypdx #2275 "Use the array methods instead of api method" :-|

:cry: indeed. Marked as a bug.  #2275 probably shouldn't have been closed, but we'll leave it in favor of this one.

**Comment by Madgvox** [comment]
I closed #2275 because it had been fixed in 0.8-edge. This is a different bug.

**Comment by evs-chris** [comment]
It looks like the binding side of this issue is fixed in edge, so I'm going to close this and link to #2420 for the observer bit.
